src/services/upload/livePhotoService.ts

```diff
@@ -0,0 +1,162 @@
+        )
+    );
+    for (let i = 0; i < mediaFiles.length - 1; i += 2) {
+        const mediaFile1 = mediaFiles[i];
+        const mediaFile2 = mediaFiles[i + 1];
+        const { fileTypeInfo: file1TypeInfo, metadata: file1Metadata } =
+            UploadService.getFileMetadataAndFileTypeInfo(mediaFile1.localID);
+        const { fileTypeInfo: file2TypeInfo, metadata: file2Metadata } =
+            UploadService.getFileMetadataAndFileTypeInfo(mediaFile2.localID);
+        if (areFilesLivePhotoAssets(mediaFile1, mediaFile2)) {
+            let imageFile;
+            let videoFile;
+            if (
+                file1TypeInfo.fileType === FILE_TYPE.IMAGE &&
+                file2TypeInfo.fileType === FILE_TYPE.VIDEO
+            ) {
+                imageFile = mediaFile1.file;
+                videoFile = mediaFile2.file;
+            } else {
+                imageFile = mediaFile2.file;
+                videoFile = mediaFile1.file;
+            }
+            const livePhotoLocalID = i;
+            analysedMediaFiles.push({
+                localID: livePhotoLocalID,
+                collectionID: mediaFile1.collectionID,
+                isLivePhoto: true,
+                livePhotoAssets: { image: imageFile, video: videoFile },
+            });
+            const livePhotoFileTypeInfo: FileTypeInfo = getLivePhotoFileType(
+                file1TypeInfo,
+                file2TypeInfo
+            );
+            const livePhotoMetadata: Metadata = getLivePhotoMetadata(
+                file1Metadata,
+                file2Metadata
+            );
+            uploadService.setFileMetadataAndFileTypeInfo(livePhotoLocalID, {
+                fileTypeInfo: { ...livePhotoFileTypeInfo },
+                metadata: { ...livePhotoMetadata },
+            });
+        } else {
+            analysedMediaFiles.push({ ...mediaFile1, isLivePhoto: false });
+            analysedMediaFiles.push({
+                ...mediaFile2,
+                isLivePhoto: false,
+            });
+        }
+    }
+    return analysedMediaFiles;
+}
+
+function areFilesLivePhotoAssets(
+    mediaFile1: FileWithCollection,
+    mediaFile2: FileWithCollection
+) {
+    const {
+        collectionID: file1collectionID,
+        file: file1,
+        localID: localID1,
+    } = mediaFile1;
+    const {
+        collectionID: file2collectionID,
+        file: file2,
+        localID: localID2,
+    } = mediaFile2;
+    const {
+        fileTypeInfo: { fileType: file1Type },
+    } = UploadService.getFileMetadataAndFileTypeInfo(localID1);
+    const {
+        fileTypeInfo: { fileType: file2Type },
+    } = UploadService.getFileMetadataAndFileTypeInfo(localID2);
+    return (
+        file1collectionID === file2collectionID &&
+        file1Type !== file2Type &&
+        file1Type !== FILE_TYPE.OTHERS &&
+        file2Type !== FILE_TYPE.OTHERS &&
+        splitFilenameAndExtension(file1.name)[0] ===
+            splitFilenameAndExtension(file2.name)[0]
```

Comment:
This check for filename can be moved to the top. We will be able to avoid calling `getFileMetadataAndFileTypeInfo` for all files.